language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/executors/CreateFunctionExecutor.java

```diff
@@ -173,11 +175,10 @@ public Object execute(ExecuteCommandContext context) throws LSCommandExecutorExc
         Optional<NonTerminalNode> enclosingNode = findEnclosingModulePartNode(fnCallExprNode.get());
         Range insertRange;
         if (enclosingNode.isPresent()) {
-            LineRange endLineRange = enclosingNode.get().lineRange();
-            newLineAtEnd = false;
-            insertRange = new Range(new Position(endLineRange.endLine().line(), endLineRange.endLine().offset()),
-                    new Position(endLineRange.endLine().line(), endLineRange.endLine().offset()));
-
+            LineRange lineRange = enclosingNode.get().lineRange();
+            newLineAtEnd = addNewLineAtEnd(enclosingNode.get());
+            insertRange = new Range(new Position(lineRange.endLine().line(), lineRange.endLine().offset()),
+                    new Position(lineRange.endLine().line(), lineRange.endLine().offset()));
```

Comment:
Shall we use `CommonUtil.toRange()` here?

Reply:
We can't use that here, as we want the Range to be starting from the endLine and offset.

Reply:
Then shall we add separate util to accept the line?

Reply:
Done